### PR TITLE
Apr 670 removing structuremap

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.RoATPService.Application.Api/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace SFA.DAS.RoATPService.Application.Api.Extensions
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static void RegisterAllTypes<T>(this IServiceCollection services, Assembly[] assemblies,
+            ServiceLifetime lifetime = ServiceLifetime.Transient)
+        {
+            var typesFromAssemblies = assemblies.SelectMany(a => a.DefinedTypes.Where(x => x.GetInterfaces().Contains(typeof(T))));
+            foreach (var type in typesFromAssemblies)
```
Comment (David-B-Read) [inline src/SFA.DAS.RoATPService.Application.Api/Extensions/ServiceCollectionExtensions.cs]: I think we should apply Dave's suggestion here too....
https://github.com/SkillsFundingAgency/das-admin-service/pull/44#discussion_r318576271

File: src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs [hunk 2]
```diff
@@ -121,38 +132,37 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
...
-                config.For<IMediator>().Use<Mediator>();
-                config.For<IDbConnection>().Use(c => new SqlConnection(Configuration.SqlConnectionString));
-
-                config.Populate(services);
-            });
-
-            return container.GetInstance<IServiceProvider>();
+            services.AddTransient<IHttpContextAccessor, HttpContextAccessor>();
+            services.AddTransient(x => Configuration);
+            services.AddTransient(c => new SqlConnection(Configuration.SqlConnectionString));
```
Comment (David-B-Read) [inline src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs]: Should we be doing this in the container if we're creating them in the repository classes?

Comment (MarkFCain) [reply src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs]: It took me ages to do it this way, I've no idea how to approach this suggestion! Let's discuss tomorrow AM.

File: src/SFA.DAS.RoATPService.Application.Api/Extensions/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.RoATPService.Application.Api/Extensions/ServiceCollectionExtensions.cs]: yeah, I'll do that

File: src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs (hunk 2 above)
Comment (David-B-Read) [reply src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs]: I think the SqlConnection registration may be redundant. It was registered in StructureMap to map to IDbConnection but not used anywhere, I don't think, as we're creating connections in the repos like this:
https://github.com/SkillsFundingAgency/das-roatp-service/blob/8b3f8e0d809aef2bb69b086fb049aaedf3b40b17/src/SFA.DAS.RoATPService.Data/LookupDataRepository.cs#L73

Comment (MarkFCain) [reply src/SFA.DAS.RoATPService.Application.Api/StartupConfiguration/Startup.cs]: Any you just mean the sqlConnection setup, right? Yeah, I think that can come out. I'll just run it and see it hangs together

**Pull request merged**